src/flint/types/fmpz_mod_poly.pyx

```diff
@@ -0,0 +1,1260 @@
+            >>> f = R([1,2,3])
+            >>> f.leading_coefficient()
+            fmpz_mod(3, 163)
+        """
+        return self[self.degree()]
+
+    def reverse(self, degree=None):
+        """
+        Return a polynomial with the coefficients of this polynomial
+        reversed.
+
+        If `degree` is not None, the output polynomial will be zero-padded
+        or truncated before being reversed. Note: degree must be non-negative.
+
+            >>> R = fmpz_mod_poly_ctx(163)
+            >>> f = R([1,2,3,4,5])
+            >>> f.reverse()
+            x^4 + 2*x^3 + 3*x^2 + 4*x + 5
+            >>> f.reverse(degree=1)
+            x + 2
+            >>> f.reverse(degree=100)
+            x^100 + 2*x^99 + 3*x^98 + 4*x^97 + 5*x^96
+        """
+        cdef fmpz_mod_poly res
+        cdef slong d
+        
+        if degree is not None:
+            d = degree
+            if d != degree or d < 0:
+                raise ValueError(f"degree argument must be a non-negative integer, got {degree}")
+        else:
+            d = fmpz_mod_poly_degree(self.val, self.ctx.mod.val)
+
+        length = d + 1
+
+        res =  fmpz_mod_poly.__new__(fmpz_mod_poly)
+        res.ctx = self.ctx
+        fmpz_mod_poly_reverse(res.val, self.val, length, self.ctx.mod.val)
+        return res
+
+    def truncate(self, slong n):
+        r"""
+        Notionally truncate the polynomial to have length `n`. If 
+        `n` is larger than the length of the input, then `self` is
+        returned. If n is not positive, then the zero polynomial 
+        is returned.
+
+        Effectively returns this polynomial :math:`\mod x^n`.
+
+            >>> R = fmpz_mod_poly_ctx(163)
+            >>> f = R([1,2,3])
+            >>> f.truncate(3) is f
+            True
+            >>> f.truncate(2)
+            2*x + 1
+            >>> f.truncate(1)
+            1
+            >>> f.truncate(0)
+            0
+            >>> f.truncate(-1)
+            0
+
+        """
+        cdef fmpz_mod_poly res
+
+        length = fmpz_mod_poly_degree(self.val, self.ctx.mod.val)
+        if n > length:
+            return self
+
+        res =  fmpz_mod_poly.__new__(fmpz_mod_poly)
+        res.ctx = self.ctx
+
+        if n <= 0:
+            return res
+
+        fmpz_mod_poly_set_trunc(res.val, self.val, n, self.ctx.mod.val)
+        return res
+
+    def is_monic(self):
+        """
+        Return whether this polynomial is monic.
+
+            >>> R = fmpz_mod_poly_ctx(163)
+            >>> x = R.gen()
+            >>> f = x**2 + 5*x + 3
+            >>> f.is_monic()
+            True
+            >>> f = 5*x**2 + x + 3
+            >>> f.is_monic()
+            False
+        """
+        return self.leading_coefficient().is_one()
+
+    def monic(self, check=True):
+        """
+        Return this polynomial divided by its leading coefficient.
+
+        If `check` is True, raises ValueError if the leading coefficient 
+        is not invertible modulo N.
```

Comment:
Does this otherwise raise a segfault?

If so then I think that it would be good to clarify that.